src/parquet/file/metadata.cc

```diff
@@ -374,10 +374,12 @@ class ColumnChunkMetaDataBuilder::ColumnChunkMetaDataBuilderImpl {
       if (properties_->version() == ParquetVersion::PARQUET_2_0) {
         thrift_encodings.push_back(ToThrift(properties_->dictionary_index_encoding()));
       }
-    }
-    if (!properties_->dictionary_enabled(column_->path()) || dictionary_fallback) {
+    } else {  // Dictionary not enabled
       thrift_encodings.push_back(ToThrift(properties_->encoding(column_->path())));
     }
+    if (dictionary_fallback) {  // Only PLAIN encoding is supported for fallback in V1
+      thrift_encodings.push_back(ToThrift(Encoding::PLAIN));
```

Comment:
hard code fallback encoding to `PLAIN` for now. For V2, we can use user specified encoding.